supabase/migrations/20260826141124_sso_providers_block_client_provider_id.sql

```
@@ -0,0 +1,100 @@
+  FROM PUBLIC;
+GRANT EXECUTE ON FUNCTION public.enforce_sso_provider_client_update_guard()
+  TO service_role;
+
+COMMENT ON FUNCTION public.enforce_sso_provider_client_update_guard() IS
+  'BEFORE INSERT/UPDATE trigger on public.sso_providers (per row). Runs on every '
+  'client write; internal roles (service_role, postgres, supabase_admin) bypass. '
+  'Client roles cannot set or change provider_id, dns_verified_at, domain, status, '
+  'or enforce_sso. Executing roles: anon, authenticated via PostgREST; internal '
+  'roles for verify-dns and /private/sso/providers create/PATCH. Table cardinality: '
+  'low per org (typically 1-5 rows); trigger touches only the inserted/updated row. '
+  'Indexes: not applicable (no table scans). Worst-case EXPLAIN (ANALYZE, BUFFERS): '
```

SQLFluff (4.3.0)

[error] Line is too long (85 > 80).

(LT05)